### PR TITLE
refactor(Core): apply clang-tidy modernize-*

### PR DESCRIPTION
**Pull request by FrancescoBorzi**
## Changes Proposed:
-  https://github.com/azerothcore/azerothcore-wotlk/discussions/9970

## Issues Addressed:

- None 

## Tests Performed:

- None

## How to Test the Changes:

These automated improvements are generally considered safe and there is no specific thing to check, but just in case:

1.  Compile and run the core
2. Access the game and try playing a little bit

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/shared/Packets/ByteBuffer.h [hunk 1]
```diff
@@ -355,7 +355,7 @@ class AC_SHARED_API ByteBuffer
         return r;
     }
 
-    template <typename T> T read(size_t pos) const
+    template <typename T> [[nodiscard]] [[nodiscard]] [[nodiscard]] [[nodiscard]] [[nodiscard]] [[nodiscard]] [[nodiscard]] [[nodiscard]] [[nodiscard]] T read(size_t pos) const
```
Comment (Winfidonarleyan) [inline src/server/shared/Packets/ByteBuffer.h:358]: ?

File: src/common/Collision/Management/MMapMgr.h [hunk 2]
```diff
@@ -50,9 +50,9 @@ namespace MMAP
 
         ~MMapData()
         {
-            for (NavMeshQuerySet::iterator i = navMeshQueries.begin(); i != navMeshQueries.end(); ++i)
+            for (auto & navMeshQuerie : navMeshQueries)
```
Comment (Kitzunu) [inline src/common/Collision/Management/MMapMgr.h:53]:
```suggestion
            for (auto& navMeshQuerie : navMeshQueries)
```

File: src/common/Utilities/EventProcessor.h [hunk 3]
```diff
@@ -43,9 +43,9 @@ class BasicEvent
 
     public:
         BasicEvent()
-            : m_abortState(AbortState::STATE_RUNNING), m_addTime(0), m_execTime(0) { }
+             = default;
```
Comment (Kitzunu) [inline src/common/Utilities/EventProcessor.h:46]: could keep this on the same line?

File: src/common/Utilities/MessageBuffer.h [hunk 4]
```diff
@@ -27,7 +27,7 @@ class MessageBuffer
     using size_type = std::vector<uint8>::size_type;
 
 public:
-    MessageBuffer() : _wpos(0), _rpos(0), _storage()
+    MessageBuffer() :  _storage()
```
Comment (Kitzunu) [inline src/common/Utilities/MessageBuffer.h:30]: why not = default?

File: src/common/Utilities/Timer.h [hunk 5]
```diff
@@ -68,8 +68,7 @@ struct IntervalTimer
 public:
     IntervalTimer()
 
-    {
-    }
+    = default;
```
Comment (Kitzunu) [inline src/common/Utilities/Timer.h:71]: can keep on the same line as IntervalTimer()?

File: src/common/Utilities/Util.h [hunk 6]
```diff
@@ -653,7 +653,7 @@ class EventMap
     typedef std::multimap<uint32, uint32> EventStore;
 
 public:
-    EventMap()  { }
+    EventMap()  = default;
```
Comment (Kitzunu) [inline src/common/Utilities/Util.h:656]: two spaces after ()

File: src/server/game/Achievements/AchievementMgr.h [hunk 7]
```diff
@@ -225,7 +225,7 @@ struct AchievementCriteriaData
 
 struct AchievementCriteriaDataSet
 {
-    AchievementCriteriaDataSet()  {}
+    AchievementCriteriaDataSet()  = default;
```
Comment (Kitzunu) [inline src/server/game/Achievements/AchievementMgr.h:228]: 2 space after ()

File: src/server/game/Battlegrounds/Battleground.h [hunk 8]
```diff
@@ -187,7 +187,7 @@ enum BattlegroundTeams
 
 struct BattlegroundObjectInfo
 {
-    BattlegroundObjectInfo()  {}
+    BattlegroundObjectInfo()  = default;
```
Comment (Kitzunu) [inline src/server/game/Battlegrounds/Battleground.h:190]: 2 spaces after ()

File: src/server/game/Battlegrounds/Battleground.h [hunk 9]
```diff
@@ -286,7 +286,7 @@ struct BattlegroundScore
 class ArenaLogEntryData
 {
 public:
-    ArenaLogEntryData()  {}
+    ArenaLogEntryData()  = default;
```
Comment (Kitzunu) [inline src/server/game/Battlegrounds/Battleground.h:289]: here too

File: src/server/game/Chat/Channels/Channel.h [hunk 10]
```diff
@@ -121,10 +122,10 @@ enum ChannelMemberFlags
 class ChannelRights
 {
 public:
-    ChannelRights() : flags(0), speakDelay(0) {}
-    ChannelRights(const uint32& f, const uint32& d, const std::string& jm, const std::string& sm, const std::set<uint32>& ml) : flags(f), speakDelay(d), joinMessage(jm), speakMessage(sm), moderators(ml) {}
-    uint32 flags;
-    uint32 speakDelay;
+    ChannelRights()  = default;
+    ChannelRights(const uint32& f, const uint32& d, std::string  jm, std::string  sm, std::set<uint32>  ml) : flags(f), speakDelay(d), joinMessage(std::move(jm)), speakMessage(std::move(sm)), moderators(std::move(ml)) {}
```
Comment (Kitzunu) [inline src/server/game/Chat/Channels/Channel.h:126]: why add 2 spaces and not 1?

File: src/server/game/DungeonFinding/LFGMgr.h [hunk 11]
```diff
@@ -18,6 +18,8 @@
 #ifndef _LFGMGR_H
 #define _LFGMGR_H
 
+#include <utility>
```
Comment (Kitzunu) [inline src/server/game/DungeonFinding/LFGMgr.h:21]: at the bottom of the includes, to follow the way other includes are done

File: src/server/game/DungeonFinding/LFGMgr.h [hunk 12]
```diff
@@ -150,8 +152,8 @@ namespace lfg
 
     struct RBEntryInfo
     {
-        RBEntryInfo() {}
-        RBEntryInfo(uint8 _roles, std::string const& _comment) : roles(_roles), comment(_comment) {}
+        RBEntryInfo() = default;
+        RBEntryInfo(uint8 _roles, std::string  _comment) : roles(_roles), comment(std::move(_comment)) {}
```
Comment (Kitzunu) [inline src/server/game/DungeonFinding/LFGMgr.h:156]: I just want to understand, why do you use std::move() in these places? What is the general purpose?

I was not going to ask but as I see it in several places I'd like to learn

File: src/server/game/DungeonFinding/LFGMgr.h [hunk 13]
```diff
@@ -389,29 +391,28 @@ namespace lfg
 
     struct LFGDungeonData
     {
-        LFGDungeonData(): id(0), name(""), map(0), type(0), expansion(0), group(0), minlevel(0),
-            maxlevel(0), difficulty(REGULAR_DIFFICULTY), seasonal(false), x(0.0f), y(0.0f), z(0.0f), o(0.0f)
+        LFGDungeonData():  name("")
```
Comment (Kitzunu) [inline src/server/game/DungeonFinding/LFGMgr.h:394]: 2 spaces after ():

File: src/server/game/DungeonFinding/LFGQueue.h [hunk 14]
```diff
@@ -18,6 +18,8 @@
 #ifndef _LFGQUEUE_H
 #define _LFGQUEUE_H
 
+#include <utility>
```
Comment (Kitzunu) [inline src/server/game/DungeonFinding/LFGQueue.h:21]: at the bottom of includes

File: src/server/game/Entities/Creature/CreatureData.h [hunk 15]
```diff
@@ -270,7 +270,7 @@ typedef std::unordered_map<uint32, EquipmentInfoContainerInternal> EquipmentInfo
 // from `creature` table
 struct CreatureData
 {
-    CreatureData()  { }
+    CreatureData()  = default;
```
Comment (Kitzunu) [inline src/server/game/Entities/Creature/CreatureData.h:273]: 2 spaces after ()

File: src/server/game/Entities/Creature/CreatureData.h [hunk 16]
```diff
@@ -429,7 +429,7 @@ typedef std::unordered_map<uint32 /*spellid*/, TrainerSpell> TrainerSpellMap;
 
 struct TrainerSpellData
 {
-    TrainerSpellData()  {}
+    TrainerSpellData()  = default;
```
Comment (Kitzunu) [inline src/server/game/Entities/Creature/CreatureData.h:432]: 2 spaces after ()

File: src/server/game/Entities/GameObject/GameObject.h [hunk 17]
```diff
@@ -720,7 +720,7 @@ enum GOState
 // from `gameobject`
 struct GameObjectData
 {
-    explicit GameObjectData()  { }
+    explicit GameObjectData()  = default;
```
Comment (Kitzunu) [inline src/server/game/Entities/GameObject/GameObject.h:723]: 2 spaces after ()

File: src/server/game/Entities/Object/Object.h [hunk 18]
```diff
@@ -655,7 +655,7 @@ class MovableMapObject
     template<class T> friend class RandomMovementGenerator;
 
 protected:
-    MovableMapObject()  {}
+    MovableMapObject()  = default;
```
Comment (Kitzunu) [inline src/server/game/Entities/Object/Object.h:658]: 2 spaces after ()

File: src/server/game/Entities/Object/ObjectGuid.h [hunk 19]
```diff
@@ -129,7 +129,7 @@ class ObjectGuid
         template<HighGuid type>
         static typename std::enable_if<ObjectGuidTraits<type>::MapSpecific, ObjectGuid>::type Create(uint32 entry, LowType counter) { return MapSpecific(type, entry, counter); }
 
-        ObjectGuid() : _guid(0) { }
+        ObjectGuid()  = default;
```
Comment (Kitzunu) [inline src/server/game/Entities/Object/ObjectGuid.h:132]: 2 spaces after ()

File: src/server/game/Entities/Object/Updates/UpdateMask.h [hunk 20]
```diff
@@ -33,9 +33,9 @@ class UpdateMask
         CLIENT_UPDATE_MASK_BITS = sizeof(ClientUpdateMaskType) * 8,
     };
 
-    UpdateMask()  { }
+    UpdateMask()  = default;
```
Comment (Kitzunu) [inline src/server/game/Entities/Object/Updates/UpdateMask.h:36]: 2 spaces after ()

File: src/server/game/Entities/Player/Player.h [hunk 21]
```diff
@@ -248,7 +248,7 @@ enum ReputationSource
 
 struct ActionButton
 {
-    ActionButton()  {}
+    ActionButton()  = default;
```
Comment (Kitzunu) [inline src/server/game/Entities/Player/Player.h:251]: 2 spaces after ()

File: src/server/game/Entities/Player/Player.h [hunk 22]
```diff
@@ -284,14 +284,14 @@ typedef std::list<PlayerCreateInfoItem> PlayerCreateInfoItems;
 
 struct PlayerClassLevelInfo
 {
-    PlayerClassLevelInfo()  {}
+    PlayerClassLevelInfo()  = default;
```
Comment (Kitzunu) [inline src/server/game/Entities/Player/Player.h:287]: 2 spaces after ()

File: src/server/game/Entities/Player/Player.h [hunk 23]
```diff
@@ -284,14 +284,14 @@ typedef std::list<PlayerCreateInfoItem> PlayerCreateInfoItems;
...
     uint16 basehealth{0};
     uint16 basemana{0};
 };
 
 struct PlayerClassInfo
 {
-    PlayerClassInfo()  { }
+    PlayerClassInfo()  = default;
```
Comment (Kitzunu) [inline src/server/game/Entities/Player/Player.h:294]: 2 spaces after ()

File: src/server/game/Entities/Player/Player.h [hunk 24]
```diff
@@ -307,7 +307,7 @@ typedef std::list<uint32> PlayerCreateInfoSpells;
 
 struct PlayerCreateInfoAction
 {
-    PlayerCreateInfoAction()  {}
+    PlayerCreateInfoAction()  = default;
```
Comment (Kitzunu) [inline src/server/game/Entities/Player/Player.h:310]: 2 spaces after ()

File: src/server/game/Entities/Player/Player.h [hunk 25]
```diff
@@ -328,7 +328,7 @@ typedef std::list<PlayerCreateInfoSkill> PlayerCreateInfoSkills;
 struct PlayerInfo
 {
     // existence checked by displayId != 0
-    PlayerInfo()  { }
+    PlayerInfo()  = default;
```
Comment (Kitzunu) [inline src/server/game/Entities/Player/Player.h:331]: 2 spaces after ()

File: src/server/game/Entities/Player/Player.h [hunk 26]
```diff
@@ -349,7 +349,7 @@ struct PlayerInfo
 
 struct PvPInfo
 {
-    PvPInfo()  {}
+    PvPInfo()  = default;
```
Comment (Kitzunu) [inline src/server/game/Entities/Player/Player.h:352]: 2 spaces after ()

File: src/server/game/Entities/Player/Player.h [hunk 27]
```diff
@@ -433,7 +433,7 @@ struct Runes
 
 struct EnchantDuration
 {
-    EnchantDuration()  {};
+    EnchantDuration()  = default;;
```
Comment (Kitzunu) [inline src/server/game/Entities/Player/Player.h:436]: 2 spaces after ()

File: src/server/game/Entities/Player/Player.h [hunk 28]
```diff
@@ -1003,7 +1003,7 @@ class Player;
 // holder for Battleground data (pussywizard: not stored in db)
 struct BGData
 {
-    BGData()  {}
+    BGData()  = default;
```
Comment (Kitzunu) [inline src/server/game/Entities/Player/Player.h:1006]: 2 spaces after ()

File: src/server/game/Entities/Unit/Unit.h [hunk 29]
```diff
@@ -357,11 +358,11 @@ enum SpellImmuneBlockType
 
 struct SpellImmune
 {
-    SpellImmune() : spellId(0), type(IMMUNITY_EFFECT), blockType(SPELL_BLOCK_TYPE_ALL) { }
+    SpellImmune()  = default;
```
Comment (Kitzunu) [inline src/server/game/Entities/Unit/Unit.h:361]: 2 spaces after ()

File: src/server/game/Entities/Unit/Unit.h [hunk 30]
```diff
@@ -1074,7 +1075,7 @@ uint32 createProcExtendMask(SpellNonMeleeDamage* damageInfo, SpellMissInfo missC
 
 struct RedirectThreatInfo
 {
-    RedirectThreatInfo()  { }
+    RedirectThreatInfo()  = default;
```
Comment (Kitzunu) [inline src/server/game/Entities/Unit/Unit.h:1078]: 2 spaces after ()

File: src/server/game/Events/GameEventMgr.h [hunk 31]
```diff
@@ -54,7 +54,7 @@ typedef std::map<uint32 /*condition id*/, GameEventFinishCondition> GameEventCon
 
 struct GameEventData
 {
-    GameEventData()  { }
+    GameEventData()  = default;
```
Comment (Kitzunu) [inline src/server/game/Events/GameEventMgr.h:57]: 2 spaces after ()

File: src/server/game/Globals/ObjectMgr.h [hunk 32]
```diff
@@ -531,7 +531,7 @@ struct PetLevelInfo
 
 struct MailLevelReward
 {
-    MailLevelReward()  {}
+    MailLevelReward()  = default;
```
Comment (Kitzunu) [inline src/server/game/Globals/ObjectMgr.h:534]: 2 spaces after ()

File: src/server/game/Globals/ObjectMgr.h [hunk 33]
```diff
@@ -622,7 +622,7 @@ struct QuestPOIPoint
     int32 x{0};
     int32 y{0};
 
-    QuestPOIPoint()  {}
+    QuestPOIPoint()  = default;
```
Comment (Kitzunu) [inline src/server/game/Globals/ObjectMgr.h:625]: 2 spaces after ()

File: src/server/game/Globals/ObjectMgr.h [hunk 34]
```diff
@@ -637,7 +637,7 @@ struct QuestPOI
     uint32 Unk4{0};
     std::vector<QuestPOIPoint> points;
 
-    QuestPOI()  {}
+    QuestPOI()  = default;
```
Comment (Kitzunu) [inline src/server/game/Globals/ObjectMgr.h:640]: 2 spaces after ()

File: src/server/game/Instances/InstanceSaveMgr.h [hunk 35]
```diff
@@ -108,7 +108,7 @@ class InstanceSaveMgr
     friend class InstanceSave;
 
 private:
-    InstanceSaveMgr()  {};
+    InstanceSaveMgr()  = default;;
```
Comment (Kitzunu) [inline src/server/game/Instances/InstanceSaveMgr.h:111]: 2 spaces after ()

File: src/server/game/Loot/LootMgr.h [hunk 36]
```diff
@@ -188,7 +188,7 @@ struct QuestItem
     bool    is_looted{false};
 
     QuestItem()
-         {}
+         = default;
```
Comment (Kitzunu) [inline src/server/game/Loot/LootMgr.h:191]: can keep on same line as questiem()

File: src/server/game/Maps/Map.h [hunk 37]
```diff
@@ -162,21 +162,21 @@ enum LiquidStatus
 
 struct LiquidData
 {
-    LiquidData() : Entry(0), Flags(0), Level(INVALID_HEIGHT), DepthLevel(INVALID_HEIGHT), Status(LIQUID_MAP_NO_WATER) { }
+    LiquidData()  = default;
```
Comment (Kitzunu) [inline src/server/game/Maps/Map.h:165]: 2 spaces after ()

File: src/server/game/Maps/Map.h [hunk 38]
```diff
@@ -162,21 +162,21 @@ enum LiquidStatus
...
+    LiquidStatus Status{LIQUID_MAP_NO_WATER};
 };
 
 struct PositionFullTerrainStatus
 {
-    PositionFullTerrainStatus() : areaId(0), floorZ(INVALID_HEIGHT), outdoors(false) { }
-    uint32 areaId;
-    float floorZ;
-    bool outdoors;
+    PositionFullTerrainStatus()  = default;
```
Comment (Kitzunu) [inline src/server/game/Maps/Map.h:176]: 2 spaces after ()

File: src/server/game/Movement/Spline/MoveSpline.h [hunk 39]
```diff
@@ -25,9 +25,9 @@ namespace Movement
 {
     struct Location : public Vector3
     {
-        Location()  {}
+        Location()  = default;
```
Comment (Kitzunu) [inline src/server/game/Movement/Spline/MoveSpline.h:28]: 2 spaces after ()

File: src/server/game/Movement/Spline/MoveSpline.h [hunk 40]
```diff
@@ -25,9 +25,9 @@ namespace Movement
...
         Location(float x, float y, float z, float o) : Vector3(x, y, z), orientation(o) {}
-        Location(const Vector3& v) : Vector3(v), orientation(0) {}
```
Comment (Kitzunu) [inline src/server/game/Movement/Spline/MoveSpline.h:30]: Why was orientation removed?

File: src/server/game/Movement/Spline/Spline.h [hunk 41]
```diff
@@ -89,7 +89,7 @@ namespace Movement
         void UninitializedSplineInitMethod(Vector3 const*, index_type, bool, index_type) { ABORT(); }
 
     public:
-        explicit SplineBase()  {}
+        explicit SplineBase()  = default;
```
Comment (Kitzunu) [inline src/server/game/Movement/Spline/Spline.h:92]: 2 spaces after ()

File: src/server/shared/DataStores/DBCStructure.h [hunk 42]
```diff
@@ -2165,7 +2165,7 @@ typedef std::map<uint32, TalentSpellPos> TalentSpellPosMap;
 
 struct TaxiPathBySourceAndDestination
 {
-    TaxiPathBySourceAndDestination()  {}
+    TaxiPathBySourceAndDestination()  = default;
```
Comment (Kitzunu) [inline src/server/shared/DataStores/DBCStructure.h:2168]: 2 spaces after ()

File: src/server/shared/DataStores/DBCStructure.h [hunk 43]
```diff
@@ -2154,7 +2154,7 @@ struct MapDifficulty
 
 struct TalentSpellPos
 {
-    TalentSpellPos()  {}
+    TalentSpellPos()  = default;
```
Comment (Kitzunu) [inline src/server/shared/DataStores/DBCStructure.h:2157]: 2 spaces after ()

File: src/server/shared/DataStores/DBCStructure.h [hunk 44]
```diff
@@ -2144,7 +2144,7 @@ struct WorldStateUI
 // Structures not used for casting to loaded DBC data and not required then packing
 struct MapDifficulty
 {
-    MapDifficulty()  {}
+    MapDifficulty()  = default;
```
Comment (Kitzunu) [inline src/server/shared/DataStores/DBCStructure.h:2147]: 2 spaces after ()

File: src/server/shared/Packets/ByteBuffer.h [hunk 45]
```diff
@@ -355,7 +355,7 @@ class AC_SHARED_API ByteBuffer
         return r;
     }
 
-    template <typename T> T read(size_t pos) const
+    template <typename T> [[nodiscard]] T read(size_t pos) const
```
Comment (Kitzunu) [inline src/server/shared/Packets/ByteBuffer.h:358]: should this be mid template?

File: src/server/shared/Realms/Realm.h [hunk 46]
```diff
@@ -36,10 +36,10 @@ enum RealmFlags
 
 struct AC_SHARED_API RealmHandle
 {
-    RealmHandle() : Realm(0) { }
+    RealmHandle()  = default;
```
Comment (Kitzunu) [inline src/server/shared/Realms/Realm.h:39]: 2 spaces after ()

File: src/server/shared/enuminfo_SharedDefines.cpp [hunk 47]
```diff
@@ -159,15 +159,15 @@ AC_API_EXPORT EnumText EnumUtils<SpellAttr0>::ToString(SpellAttr0 value)
     {
         case SPELL_ATTR0_PROC_FAILURE_BURNS_CHARGE: return { "SPELL_ATTR0_PROC_FAILURE_BURNS_CHARGE", "Unknown attribute 0@Attr0", "" };
         case SPELL_ATTR0_USES_RANGED_SLOT: return { "SPELL_ATTR0_USES_RANGED_SLOT", "Treat as ranged attack", "Use ammo, ranged attack range modifiers, ranged haste, etc." };
-        case SPELL_ATTR0_ON_NEXT_SWING_NO_DAMAGE: return { "SPELL_ATTR0_ON_NEXT_SWING_NO_DAMAGE", "On next melee (type 1)", "Both \042on next swing\042 attributes have identical handling in server & client" };
+        case SPELL_ATTR0_ON_NEXT_SWING_NO_DAMAGE: return { "SPELL_ATTR0_ON_NEXT_SWING_NO_DAMAGE", "On next melee (type 1)", R"(Both "on next swing" attributes have identical handling in server & client)" };
```
Comment (Kitzunu) [inline src/server/shared/enuminfo_SharedDefines.cpp:162]: ?

**Comment by FrancescoBorzi** [comment]
@Kitzunu thanks for your comments, however all of this was auto-generated. I'd keep it as it is and address the minor codestyle improvements in separate PR(s), otherwise it will be too much work.

**Comment by Kitzunu** [comment]
@FrancescoBorzi  https://github.com/azerothcore/azerothcore-wotlk/pull/9975#discussion_r779111290 atleast this? :D

File: src/server/shared/enuminfo_SharedDefines.cpp (hunk 47 above)
Comment (Kitzunu) [reply src/server/shared/enuminfo_SharedDefines.cpp:162]: What is R?

**Comment by FrancescoBorzi** [comment]
> 

ok, feel free to edit my branch please

**Comment by Kitzunu** [comment]
I still want to know what is R

**Comment by Winfidonarleyan** [comment]
> I still want to know what is R

hah, check this https://github.com/TrinityCore/TrinityCore/commit/849c906014b98f59652cc0f761ada709b92e2f18

**Comment by FrancescoBorzi** [comment]
Tested by @IntelligentQuantum 

```
AzerothCore rev. cd3dd437cb3b 2022-01-14 21:23:24 +0100 (clang-tidy-modernise-2 branch) (Unix, RelWithDebInfo, Static)
Connected players: 1. Characters in world: 1.
Connection peak: 1.
Server uptime: 48 minute(s) 14 second(s).
Update time diff: 10ms, average: 13ms.

Tested some spell, bg, lfg, swim flight etc...
```

**Review by Nyeriah — APPROVED** [review]

**Comment by Nyeriah** [comment]
/home/runner/work/azerothcore-wotlk/azerothcore-wotlk/src/server/game/Maps/ZoneScript.h:32:111: fatal error: no member named 'id1' in 'CreatureData'
    virtual uint32 GetCreatureEntry(ObjectGuid::LowType /*guidlow*/, CreatureData const* data) { return data->id1; }
                                                                                                        ~~~~  ^
1 error generated.

**Pull request merged**